src/main/java/com/microsoft/azure/functions/worker/broker/JavaFunctionBroker.java

```diff
@@ -87,14 +93,48 @@ private void addSearchPathsToClassLoader(FunctionMethodDescriptor function) thro
 		function.getLibDirectory().ifPresent(d -> registerWithClassLoaderProvider(d));
 	}
 
-	private void registerWithClassLoaderProvider(File libDirectory) {
+	void registerWithClassLoaderProvider(File libDirectory) {
 		try {
-			classLoaderProvider.addDirectory(libDirectory);
+			if(SystemUtils.IS_JAVA_1_8) {
+				String workerLibPath = System.getenv(Constants.FUNCTIONS_WORKER_DIRECTORY) + "/lib";
+				String javaReverseLibLoading = System.getenv(Constants.FUNCTIONS_WORKER_JAVA_LOAD_APP_LIBS);
+				File workerLib = new File(workerLibPath);
+				if(Utilities.isTrue(javaReverseLibLoading)) {
+					if(verifyLibrariesExist (workerLib, workerLibPath, "CLIENT_LIB_FIRST")) {
+						classLoaderProvider.addDirectory(libDirectory);
+						classLoaderProvider.addDirectory(workerLib);
+					}
+				} else {
+					if(verifyLibrariesExist (workerLib, workerLibPath, "WORKER_LIB_FIRST")) {
```

Comment:
Change to
```java
if(SystemUtils.IS_JAVA_1_8) {
				String workerLibPath = System.getenv(Constants.FUNCTIONS_WORKER_DIRECTORY) + "/lib";
				String javaReverseLibLoading = System.getenv(Constants.FUNCTIONS_WORKER_JAVA_LOAD_APP_LIBS);
				File workerLib = new File(workerLibPath);
                verifyLibrariesExist (workerLib, workerLibPath);
				if(Utilities.isTrue(javaReverseLibLoading)) {
						classLoaderProvider.addDirectory(libDirectory);
						classLoaderProvider.addDirectory(workerLib);
					}else {
						classLoaderProvider.addDirectory(workerLib);
						classLoaderProvider.addDirectory(libDirectory);
					}
				}
```

Reply:
move this to a different method. Make it much easier to read code. Move `SystemUtils.IS_JAVA_1_8` to utilities as well : `IsJava8`

Reply:
Please have a look on the next code update, I believe it is more clean. I just prefer to be in one place for more easy to follow. And Util class has only generic methods.